src/cryptonote_protocol/quorumnet.cpp

```diff
@@ -1659,11 +1657,8 @@ void handle_pulse_participation_bit_or_bitset(Message &m, QnetState& qnet, bool
     msg.type = pulse::message_type::handshake;
   }
 
-  {
-    std::unique_lock lock{qnet.pulse_message_queue_mutex};
-    qnet.pulse_message_queue.push(std::move(msg));
-  }
-  qnet.pulse_message_queue_cv.notify_one();
+  auto *self = reinterpret_cast<void *>(&qnet);
```

Comment:
Conversion to `void*` is implicitly allowed from any pointer type, so you don't need a `reinterpret_cast` here:

```c++
void *self = &qnet;
```
or you could drop it entirely and just capture and reference it in the lambda:
```C++
qnet.lmq.job([&qnet, data = std::move(msg)]() { pulse::handle_message(&qnet, data); }, qnet.core.pulse_thread_id());
```

Reply:
Updated